code/hsec-core/src/Security/Advisories/Core/Advisory.hs

```diff
@@ -44,10 +48,30 @@ data Advisory = Advisory
   }
   deriving stock (Show)
 
+data Ecosystem = Hackage Text | GHC GHCComponent
+  deriving stock (Show, Eq)
+
+-- Keep this list in sync with the 'ghcComponentFromText' below
+data GHCComponent = GHCCompiler | GHCi | GHCRTS
+  deriving stock (Show, Eq)
+
+ghcComponentToText :: GHCComponent -> Text
+ghcComponentToText c = case c of
+  GHCCompiler -> "compiler"
+  GHCi -> "ghci"
+  GHCRTS -> "rts"
+
+ghcComponentFromText :: Text -> Maybe GHCComponent
+ghcComponentFromText c = case c of
+  "compiler" -> Just GHCCompiler
+  "ghci" -> Just GHCi
+  "rts" -> Just GHCRTS
+  _ -> Nothing
```

Comment:
This is good.  We should also file a PR to osv-schema to document these values (I will do this now).

Reply:
For completeness, we should probably also add:
- `ghc-pkg`
- `runghc`
- `iserv`
- `hp2ps`
- `hpc` (only old versions on hackage)
- `hsc2hs` (recent versions are on hackage but it is also in the GHC distribution so we should be able to represent that).
- `haddock` (see `hsc2hs`)